examples/pxScene2d/src/rcvrcore/optimus.js

```diff
@@ -408,16 +482,28 @@ function Application(props) {
     _externalApp.ready.then(function() {
     }, function rejection() {
       _this.log("failed to launch Spark instance app: " + _this.id);
-      _readyReject(new Error("failed to create"));
+      _readyBaseReject(new Error("failed to create"));
+      _uiReadyReject();
       _this.applicationClosed();
     });
     _externalApp.remoteReady.then(function() {
       _this.log("spark instance ready");
-      _readyResolve();
+      
+      _externalApp.api.on("onApplicationLoaded", function(e)
+      {
+        if(e.success)
+          _urlChangeResolve();
+        else
+          _urlChangeReject();
+      });
+      
+      _readyBaseResolve();
+      _uiReadyResolve();
```

Comment:
Is it possible for these callback methods to be undefined or null at this point?  If so, let's add a check to make sure they're valid before calling the functions.